File: BotsSDK/widgetsdk/src/main/java/com/kore/ai/widgetsdk/views/viewutils/RoundedCornersTransform.java
package com.kore.ai.widgetsdk.views.viewutils;

/**
 * Created by Pradeep Mahato on 22/7/17.
 * Copyright (c) 2014 Kore Inc. All rights reserved.
 */
import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.RectF;

import com.squareup.picasso.Transformation;


public class RoundedCornersTransform implements Transformation {
    public void setR(float r) {
        this.r = r;
    }

    private float r = 0;
    @Override
    public Bitmap transform(Bitmap source) {
        try {
            int size = Math.min(source.getWidth(), source.getHeight());

            int x = (source.getWidth() - size) / 2;
            int y = (source.getHeight() - size) / 2;

            Bitmap squaredBitmap = Bitmap.createBitmap(source, x, y, size, size);
            if (squaredBitmap != source) {
                source.recycle();
            }
            Bitmap.Config config = source.getConfig() != null ? source.getConfig() : Bitmap.Config.ARGB_8888;

            Bitmap bitmap = Bitmap.createBitmap(size, size, config);

            Canvas canvas = new Canvas(bitmap);
            Paint paint = new Paint();
            BitmapShader shader = new BitmapShader(squaredBitmap, BitmapShader.TileMode.CLAMP, BitmapShader.TileMode.CLAMP);
            paint.setShader(shader);
            paint.setAntiAlias(true);
            if (r == 0) {
                r = size / 8f;
            }
            canvas.drawRoundRect(new RectF(0, 0, source.getWidth(), source.getHeight()), r, r, paint);
            squaredBitmap.recycle();
            return bitmap;
        }catch (Exception e){
            e.printStackTrace();
        }
        return null;
    }

    @Override
    public String key() {
        return "rounded_corners";
    }
}

File: BotSdk/ui/src/main/java/com/kore/ui/transforms/RoundedCornersTransform.kt
package com.kore.ui.transforms

import android.graphics.Bitmap
import android.graphics.BitmapShader
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.RectF
import android.graphics.Shader
import com.squareup.picasso.Transformation
import androidx.core.graphics.createBitmap

class RoundedCornersTransform : Transformation {
    fun setR(r: Float) {
        this.r = r
    }

    private var r = 0f
    override fun transform(source: Bitmap): Bitmap? {
        try {
            val size = source.width.coerceAtMost(source.height)
            val x = (source.width - size) / 2
            val y = (source.height - size) / 2
            val squaredBitmap = Bitmap.createBitmap(source, x, y, size, size)
            if (squaredBitmap != source) source.recycle()
            val config: Bitmap.Config? = source.config
            val bitmap = createBitmap(size, size, config ?: Bitmap.Config.ARGB_8888)
            val canvas = Canvas(bitmap)
            val paint = Paint()
            val shader = BitmapShader(squaredBitmap, Shader.TileMode.CLAMP, Shader.TileMode.CLAMP)
            paint.shader = shader
            paint.isAntiAlias = true
            if (r == 0f) {
                r = size / 8f
            }
            canvas.drawRoundRect(RectF(0f, 0f, source.width.toFloat(), source.height.toFloat()), r, r, paint)
            squaredBitmap.recycle()
            return bitmap
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return null
    }

    override fun key(): String {
        return "rounded_corners"
    }
}